src/com/nicoles_change_maker/ChangeMaker.java

```diff
@@ -0,0 +1,61 @@
+package com.nicoles_change_maker;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Iterator;
+/**
+ * Created by bears8yourface on 1/5/16.
+ */
+public class ChangeMaker {
+    double remainingDollars;
+
+    public HashMap makeChange(double dollarAmount) {
+        remainingDollars = dollarAmount;
+
+        if (checkDollarAmountIsValid(dollarAmount)) {
+            return evaluateEachCoinAmount();
+        }
+        throw new IllegalArgumentException();
+    }
+
+    private HashMap evaluateEachCoinAmount() {
+        HashMap changeCollection = new HashMap();
+        Iterator coinIterator = new CoinMap().getIterator();
+
+        while(coinIterator.hasNext()) {
+            Map.Entry coinValueMapping = (Map.Entry) coinIterator.next();
+            evaluateCoinAmountForInclusionInChangeCollection(changeCollection, coinValueMapping);
+        }
+
+        return changeCollection;
+    }
+
+    private void evaluateCoinAmountForInclusionInChangeCollection(HashMap changeCollection, Map.Entry coinValueMapping) {
```

Comment:
:+1: for being unafraid of long names. But I'm not sure the length helps reveal the intention. Again, what does "evaluating" a coin mean, and what does it mean to "include a coin in a change collection". Plain language names are OK even if they sound goofy.